### PR TITLE
[INLONG-6264][Sort] Support TiDB

### LINKED ISSUE #6264: [Feature][Sort] Support TiDB
### Description

Support TiDB

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by chestnut-c**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #6264

### Motivation

Buiold TiDB Connector

### Modifications

Support TiDB 

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TidbExtractNode.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    @Override
+    public String getPrimaryKey() {
+        return primaryKey;
+    }
+
+    @Override
+    public Map<String, String> tableOptions() {
+        Map<String, String> options = super.tableOptions();
+
+        options.put("connector", "tidb");
```
Comment (yunqingmoswu) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TidbExtractNode.java:143]: Maybe it is better to extract constants for the constant string?

File: inlong-sort/sort-connectors/tidb/pom.xml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            <artifactId>commons-beanutils</artifactId>
+            <version>${dep.apache.commons.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>${dep.guava.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/tidb/pom.xml:95]: Is this dependency require?

File: inlong-sort/sort-connectors/tidb/pom.xml [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            <artifactId>tikv-client-java</artifactId>
+            <version>${dep.tikv-client.version}</version>
+        </dependency>
+        <!-- apache commons -->
+        <dependency>
+            <groupId>commons-beanutils</groupId>
+            <artifactId>commons-beanutils</artifactId>
+            <version>${dep.apache.commons.version}</version>
+        </dependency>
+        <dependency>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/tidb/pom.xml:89]: Is this dependency require?

File: inlong-sort/sort-connectors/tidb/pom.xml [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            <version>${dep.jackson.version}</version>
+        </dependency>
+        <!-- tikv client -->
+        <dependency>
+            <groupId>org.tikv</groupId>
+            <artifactId>tikv-client-java</artifactId>
+            <version>${dep.tikv-client.version}</version>
+        </dependency>
+        <!-- apache commons -->
+        <dependency>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/tidb/pom.xml:84]: Is this dependency require?

File: inlong-sort/sort-connectors/tidb/pom.xml [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+            <scope>${flink.kafka.connector.scope}</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-table-api-java-bridge_${dep.scala.binary.version}</artifactId>
+            <version>${dep.flink-base.version}</version>
+            <scope>${flink-base.scope}</scope>
+        </dependency>
+        <!-- jackson -->
+        <dependency>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/tidb/pom.xml:72]: Is this dependency require?

File: inlong-sort/sort-connectors/tidb/pom.xml [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+            <version>${dep.flink.version}</version>
+            <scope>${flink.jdbc.connector.scope}</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-connector-kafka_${dep.scala.binary.version}</artifactId>
+            <version>${dep.flink.version}</version>
+            <scope>${flink.kafka.connector.scope}</scope>
+        </dependency>
+        <dependency>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/tidb/pom.xml:65]: Is this dependency require?

File: inlong-sort/sort-connectors/tidb/pom.xml [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+
+    <dependencies>
+        <!-- flink -->
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-connector-jdbc_${dep.scala.binary.version}</artifactId>
+            <version>${dep.flink.version}</version>
+            <scope>${flink.jdbc.connector.scope}</scope>
+        </dependency>
+        <dependency>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/tidb/pom.xml:59]: Is this dependency require?

File: inlong-sort/sort-connectors/tidb/pom.xml [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+        <flink-base.scope>provided</flink-base.scope>
+        <mysql.driver.scope>compile</mysql.driver.scope>
+        <flink.jdbc.connector.scope>compile</flink.jdbc.connector.scope>
+        <flink.kafka.connector.scope>compile</flink.kafka.connector.scope>
+    </properties>
+
+    <dependencies>
+        <!-- flink -->
+        <dependency>
```
Comment (yunqingmoswu) [inline inlong-sort/sort-connectors/tidb/pom.xml:53]: Is this dependency require?

File: inlong-sort/sort-connectors/tidb/pom.xml (hunk 2 above)
Comment (chestnut-c) [reply inlong-sort/sort-connectors/tidb/pom.xml:95]: yes

File: inlong-sort/sort-connectors/tidb/pom.xml (hunk 4 above)
Comment (chestnut-c) [reply inlong-sort/sort-connectors/tidb/pom.xml:84]: yes

File: inlong-sort/sort-connectors/tidb/pom.xml (hunk 5 above)
Comment (chestnut-c) [reply inlong-sort/sort-connectors/tidb/pom.xml:72]: yes

File: inlong-sort/sort-connectors/tidb/pom.xml (hunk 6 above)
Comment (chestnut-c) [reply inlong-sort/sort-connectors/tidb/pom.xml:65]: yes

File: inlong-sort/sort-connectors/tidb/pom.xml (hunk 7 above)
Comment (chestnut-c) [reply inlong-sort/sort-connectors/tidb/pom.xml:59]: yes

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TidbExtractNode.java (hunk 1 above)
Comment (chestnut-c) [reply inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/extract/TidbExtractNode.java:143]: ok i will change it

File: inlong-sort/sort-connectors/tidb/pom.xml (hunk 3 above)
Comment (chestnut-c) [reply inlong-sort/sort-connectors/tidb/pom.xml:89]: yes Some operations are used

**Comment by yunqingmoswu** [comment]
Please handle the conflicts.

**Pull request closed without merging**